### DISCUSSION TITLE
create_index能否重复调用？会不会有什么问题？

### DISCUSSION THREAD

**Discussion by jilieryuyi** [milvus-io/milvus#7287] [Q&A and General discussion]
如题

**Comment by yhmo** [comment]
- create_index() will rebuild index **if you change the index parameters**, such as index type, nlist. This behavior is both in v1.x and v2.0.

- **If the index parameters are always same**, frequently call create_index() will lead to lot of small segments.

**Reply by jilieryuyi** [reply]
thank you, got it

**Reply by jilieryuyi** [reply]
> * create_index() will rebuild index **if you change the index parameters**, such as index type, nlist. This behavior is both in v1.x and v2.0.
> * **If the index parameters are always same**, frequently call create_index() will lead to lot of small segments.

有什么好的建议，比如一直写数据的时候，为了保证查询效率，多少数据量调用下创建索引比较合适，还是不要调用，直接使用配置形式的自动化生成索引？

**Reply by yhmo** [reply]
2.0建议让它内部自动生成索引吧。
理想的做法是，对于正在growing的segment，应该有一个临时索引的机制来加速查询，不过现在还没有做这件事。
